src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java

```diff
@@ -125,21 +134,29 @@ EurekaOneDiscoveryStrategy build() {
     private static final int DISCOVERY_RETRY_TIMEOUT = 1;
 
     private final EurekaClient eurekaClient;
+    private final String groupName;
     private final ApplicationInfoManager applicationInfoManager;
 
     private final Boolean useClasspathEurekaClientProps;
     private final String namespace;
     private StatusChangeStrategy statusChangeStrategy;
+    private final Boolean skipEurekaRegistrationVerification;
+    private final Boolean useMetadataForHostAndPort;
 
     private EurekaOneDiscoveryStrategy(final EurekaOneDiscoveryStrategyBuilder builder) {
         super(builder.logger, builder.properties);
 
         this.namespace = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, NAMESPACE, "hazelcast");
         boolean selfRegistration = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, SELF_REGISTRATION, true);
+        this.useMetadataForHostAndPort = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, USE_METADATA_FOR_HOST_AND_PORT, false);
+        this.skipEurekaRegistrationVerification = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, SKIP_EUREKA_REGISTRATION_VERIFICATION, false);
         this.useClasspathEurekaClientProps = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, USE_CLASSPATH_EUREKA_CLIENT_PROPS, true);
+        this.groupName = builder.groupName != null ? builder.groupName : GroupConfig.DEFAULT_GROUP_NAME;
         // override registration if requested
-        if (!selfRegistration) {
+        if (!selfRegistration && !useMetadataForHostAndPort) {
             statusChangeStrategy = new NoopUpdater();
+        } else if (useMetadataForHostAndPort) {
+            statusChangeStrategy = new MetadataUpdater(builder.discoveryNode, selfRegistration, this.groupName);
```

Comment:
if `selfRegistration` is `false` and `useMetadataForHostAndPort` is `true`, the latter will override the former and self-register the instance anyway... am I reading it correctly?

Reply:
In current solution this will update metadata but will not change Eureka status. See `MetadataUpdater`. Maybe we should skip metadata registration if `selfRegistration` is `false`.

Reply:
Yes, I think it would be more user-friendly to skip registration if `selfRegistration` is false.

Reply:
Updated.

Reply:
I find this if/else if/else a bit confusing to read (it's probably just me...). We have three cases here:

1. use whatever change strategy was defined in the builder
2. no-op
3. use-metadata

And 2 and 3 may override 1. 

Is this correct / covers all cases?

Reply:
Correct. To be honest I'm not the author of this block and I'm also a little bit confused. 
My idea is to move this logic to `EurekaOneDiscoveryStrategyBuilder.build()` method and always use strategy defined in the builder.

Reply:
But this is not so easy because `EurekaOneDiscoveryStrategyBuilder` can't easily read Hazelcast properties...

Reply:
My idea is to remove logic from `EurekaOneDiscoveryStrategyBuilder.build()` and use following code in `EurekaOneDiscoveryStrategy` constructor:
```java
if (builder.changeStrategy != null) {
	this.statusChangeStrategy = builder.changeStrategy;
} else if ((!selfRegistration && !useMetadataForHostAndPort) || builder.discoveryNode == null) {
	statusChangeStrategy = new NoopUpdater();
} else if (selfRegistration && !useMetadataForHostAndPort) {
	this.statusChangeStrategy = new DefaultUpdater();
} else if (useMetadataForHostAndPort) {
	statusChangeStrategy = new MetadataUpdater(builder.discoveryNode, selfRegistration, this.groupName);
} else {
	throw new IllegalStateException("Can't configure StatusChangeStrategy");
}
```

Reply:
_ Maybe we should skip metadata registration if `selfRegistration` is `false`._

I had to revert this change. I missed case important for me. When we reuse existing Eureka client I want to update Eureka metadata map but I don't want to update Eureka status. I'm not sure if we should support other cases.